### PR TITLE
[JENKINS-49905] - verify-signature in core/pom.xml

### PR DESCRIPTION
**Pull request by odra**
See [JENKINS-49905](https://issues.jenkins-ci.org/browse/JENKINS-49905).

### Proposed changelog entries

* Internal: Adds verify-signature test in core/pom.xml file

### Submitter checklist

- [x] JIRA issue is well described
- [x] Changelog entry appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
      * Use the `Internal: ` prefix if the change has no user-visible impact (API, test frameworks, etc.)
- [x] Appropriate autotests or explanation to why this change has no tests
- [x] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@oleg-nenashev

### CODE REVIEW

**Review by oleg-nenashev — CHANGES_REQUESTED** [review]
Thanks for the proposal @odra !

With the current configuration the release will be doable only on machines which a capable of releasing Remoting.

You can probably just remove the keystore specification:

```
<alias>${hudson.sign.alias}</alias>
<storepass>${hudson.sign.storepass}</storepass>
<keystore>${hudson.sign.keystore}</keystore>
<storetype>${hudson.sign.storetype}</storetype>              <providerClass>${hudson.sign.providerClass</providerClass>
<providerArg>${hudson.sign.providerArg</providerArg>
```

It actually should not be needed even in Remoting pom.xml

**Comment by odra** [comment]
@oleg-nenashev  Should there be changes in the Jenkinsfile to pass those paramaters in case of a release build?

**Comment by jeffret-b** [comment]
Who is the intended user for this change?

**Comment by odra** [comment]
It improves the internal release process so it will fail if using an expired cert for release builds, so the intended user in this case would be jenkins core developer(s)?

**Comment by oleg-nenashev** [comment]
> Should there be changes in the Jenkinsfile to pass those paramaters in case of a release build?

Release builds happens outside Jenkins so far. 
@olblak has recently started a thread about changing it tho: https://groups.google.com/forum/#!topic/jenkinsci-dev/Og0N2SO76eY

**Review by oleg-nenashev — APPROVED** [review]
Test failure is unrelated. The change itself looks good to me as long as it stays in the release profile

CC @daniel-beck who may have some insights about @kohsuke 's release flow

**Comment by oleg-nenashev** [comment]
@daniel-beck it's fixed

File: core/pom.xml
Comment (oleg-nenashev) [inline core/pom.xml]: not required as well. `verify-signature` should be enough

File: core/pom.xml
Comment (oleg-nenashev) [inline core/pom.xml]: hm?

Comment (odra) [reply core/pom.xml]: I was getting an error from maven (probably due to my maven version), let me remove it.

**Review by jeffret-b — APPROVED** [review]
It looks fine to me, but before merging it I would want to be more clear on how it would be used. I think the potential impact is to Kohsuke as he prepares releases so we would want to make sure he is at least aware of the change and better that he approves of it. If this change is intended as part of a larger effort to automate Jenkins releases then perhaps this change should be held until that proposal is further along.

**Comment by odra** [comment]
@jeffret-b is there a jep or a ticket collection that I could take a look? I Would be happy to help in the release automation work.

**Comment by oleg-nenashev** [comment]
I have asked for a review from Kohsuke. On hold for now

**Review by kohsuke — APPROVED** [review]
Thanks for the heads up. Let's merge this!

I can't think of any reasons why this wouldn't work, and if something goes wrong I know what to think about!

**Comment by oleg-nenashev** [comment]
Thanks @kohsuke! 
BTW, we could move the code signing check to the main pom.xml so that all artifacts are checked.
I will test it locally before the weekly

**Comment by oleg-nenashev** [comment]
Thanks a lot for your contribution @odra!

**Comment by oleg-nenashev** [comment]
According to @kohsuke, it didn't work for him: https://github.com/jenkinsci/jenkins/commit/900fab5bd82feb09b1089427034223e54b4dfdd6 

It seems signing verification was invoked before the signing actually happened

**Comment by odra** [comment]
Tried pinging people in irc but got no response, Is there a documentation somewhere about the release process so I can reproduce the issue and pr a fix?

**Pull request merged**
